vowpalwabbit/global_data.h

```diff
@@ -572,6 +580,17 @@ struct GlobalState
    * feature related setting?
    */
   uint64_t parse_mask;  // 1 << num_bits -1
+
+  /*
+   * following 7 are feature related settings?
+   * not so sure but feels like its state, related to features though
+   */
+  // Referenced by examples as their set of interactions. Can be overriden by reductions.
+  std::vector<std::vector<namespace_index>> interactions;
+
+  /* highly related to feature.ignore but this is just evil state*/
```

Comment:
could be a code smell, in reality we care about ignoring if we have things to ignore